addons/binding/org.openhab.binding.smartthings/src/main/java/org/openhab/binding/smartthings/discovery/SmartthingsDiscoveryService.java

```diff
@@ -0,0 +1,244 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.smartthings.discovery;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.regex.Pattern;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.smartthings.SmartthingsBindingConstants;
+import org.openhab.binding.smartthings.handler.SmartthingsBridgeHandler;
+import org.openhab.binding.smartthings.internal.SmartthingsHttpClient;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * Smartthings Discovery service
+ *
+ * @author Bob Raker - Initial contribution
+ *
+ */
+public class SmartthingsDiscoveryService extends AbstractDiscoveryService implements EventHandler {
+
+    private static final int SEARCH_TIME = 30;
+    private static final int INITIAL_DELAY = 5;
+    private static final int SCAN_INTERVAL = 180;
+
+    private final Pattern findIllegalChars = Pattern.compile("[^A-Za-z0-9_-]");
+
```

Comment:
You love newlines, I see. But also think about us reviewers. You binding is 5000 LOC, and I bet 500 lines are "unnecessary" (subjective) newlines.

If possible can you group fields and remove newlines between single entries? Usually final fields are one group, one-time initialized and runtime fields are groups.